doc/src/challenges/XXXX-rc.md

```diff
@@ -0,0 +1,130 @@
+# Challenge XXXX[^challenge_id]: Verify reference-counted Cell implementation
+
+- **Status:** Open
+- **Solution:** *Option field to point to the PR that solved this challenge.*
+- **Tracking Issue:** *Link to issue*
+- **Start date:** 2025/06/01
+- **End date:** 2025/12/31
+- **Reward:** *TBD*[^reward]
+
+-------------------
+
+
+## Goal
+
+The goal of this challenge is to verify Rc and its related Weak implementation. Rc is the library-provided building block that enables safe multiple ownership of data through reference counting for single-threaded cases, as opposed to the usual ownership types used by Rust. A related challenge verifies the Arc implementation, which is atomic multi-threaded.
+
+## Motivation
+
+The Rc (for single-threaded code) and Arc (atomic multi-threaded) cell types are widely used in Rust programs to enable shared ownership of data through reference counting. Since shared ownership is generally not permitted by Rust's type system, these implementations use unsafe code to bypass Rust's usual compile-time checks. Verifying the Rust standard library thus fundamentally requires verification of these types.
```

Comment:
In terms of reading it is a bit awkward that only the second paragraph explains what `Rc` and `Arc` stand for. Please make sure you include the explanation of the acronyms upon first use.